microovn/frr/bgp/redirect.go

```diff
@@ -288,6 +298,39 @@ func createVrf(ctx context.Context, s state.State, extConnections []types.BgpExt
 	return nil
 }
 
+func generateVeth(ctx context.Context, _ state.State, interfaceName string, macs []string, tableID string) error {
+	var confBuilder strings.Builder
+
+	vethname := getBgpVethName(interfaceName)
+	vrfName := getVrfName(tableID)
+	// Ensure we don't announce any default route from VRF to our peer.
+	fmt.Fprint(&confBuilder, "network:\n")
+	fmt.Fprint(&confBuilder, "  version: 2\n")
+	fmt.Fprint(&confBuilder, "  virtual-ethernets:\n")
+
+	fmt.Fprintf(&confBuilder, "    %s-a:\n      peer: %s-b\n", vethname, vethname)
+	if len(macs) >= 1 {
+		fmt.Fprintf(&confBuilder, "      macaddress: %s\n", macs[0])
+	}
+
+	fmt.Fprintf(&confBuilder, "    %s-b:\n      peer: %s-a\n", vethname, vethname)
+	if len(macs) >= 2 {
+		fmt.Fprintf(&confBuilder, "      macaddress: %s\n", macs[1])
+	}
+
+	if tableID != "" {
+		fmt.Fprintf(&confBuilder, "  vrfs:\n    %s:\n      table: %s\n      interfaces:\n        - %s-a\n", vrfName, tableID, vethname)
+	}
```

Comment:
The go [yaml package](https://pkg.go.dev/gopkg.in/yaml.v3) allows seamless marshaling of structs into valid YAML.

Let's make use that instead of patching strings together.